File: packages/emitter/testcases/common/attributes/basic/Attributes.ts
// Test: declaration-level attributes using A<T>().add(...) pattern
import { attributes as A } from "@tsonic/core/lang.js";
import {
  SerializableAttribute,
  ObsoleteAttribute,
} from "@tsonic/dotnet/System.js";

// Class with single attribute
export class User {
  name!: string;
  age!: number;
}
A<User>().add(SerializableAttribute);

// Class with attribute and positional argument
export class Config {
  setting!: string;
}
A<Config>().add(ObsoleteAttribute, "Use NewConfig instead");

// Class with multiple attributes
export class AnnotatedService {
  data!: string;
}
A<AnnotatedService>().add(SerializableAttribute);
A<AnnotatedService>().add(ObsoleteAttribute, "Deprecated");

File: packages/emitter/testcases/common/dotnet/attributes/basic/Attributes.cs
namespace TestCases.common.attributes.basic
{
    [SerializableAttribute]
    public class User
    {
        public string name;

        public double age;
    }
    [ObsoleteAttribute("Use NewConfig instead")]
    public class Config
    {
        public string setting;
    }
    [SerializableAttribute]
    [ObsoleteAttribute("Deprecated")]
    public class LegacyService
    {
        public string data;
    }
}